script.tag-generator/main.py

```diff
@@ -0,0 +1,643 @@
+            newimdbtag = imdbtaglist[scrapecount]
+            imdblist, imdbuser = scrapeimdb(imdburl, scrapecount)
+            moviecount = moviecount + writeimdbtags(imdblist, medialist, newimdbtag)
+            scrapecount = scrapecount + 1
+    else:
+        debuglog("TAG-GEN: Skipping IMDb tag writing.")
+        moviecount = 0
+
+    # Trakt movies tag writing
+    if c_usetrakt and not wipeout and (len(c_trakt_token) == 64):
+        if c_notify:
+            notify(_getstr(30037))
+        xbmc.log(msg="TAG-GEN: Starting Trakt tag writing.", level=xbmc.LOGNOTICE)
+        i = 0
+        for this_trakt_list in trakt_lists:
+            this_trakt_tag = trakt_tags[i]
+            this_trakt_user = trakt_users[i]
+            try:
+                trakt_movies = get_trakt_movies(this_trakt_user, this_trakt_list, c_trakt_token)
+                moviecount += write_trakt_tags(trakt_movies, medialist, this_trakt_tag)
+            except:
+                xbmc.log(msg="TAG-GEN: Could not retrieve movies from Trakt API.", level=xbmc.LOGERROR)
+            i += 1
+    else:
+        debuglog("TAG-GEN: Skipping Trakt tag writing.")
+
+    # Stand-up Comedy tag writing
+    if c_usestandup and not wipeout:
+        if c_notify:
+            notify(_getstr(30038))
+        newwikitag = c_standuptag
+        xbmc.log(msg="TAG-GEN: Starting stand-up tag writing.", level=xbmc.LOGNOTICE)
+        comedians = scrapewiki()
+        comiccount = writestanduptags(comedians, medialist, newwikitag)
+    else:
+        debuglog("TAG-GEN: Skipping stand-up tag writing.")
+
+    # Awards tag writing
+    if c_useawards and not wipeout and (c_oscarwinners or c_oscarnominees):
+        if c_notify:
+            notify(_getstr(30039))
+        medialist = getkodidb()
+        xbmc.log("TAG-GEN: Starting awards tag writing.", level=xbmc.LOGNOTICE)
+        awardscount = write_award_tags(medialist, c_oscarwinners, c_oscarnominees)
+    else:
+        debuglog("TAG-GEN: Skipping awards tag writing.")
+
+    if manual:
+        dialog = xbmcgui.Dialog()
+        try:
+            _ = awardscount
+        except:
+            awardscount = 0
+        ok = dialog.ok("Tag Generator", _getstr(30031) + str(moviecount) + _getstr(30032) + str(comiccount)
+                       + _getstr(30033) + str(awardscount) + _getstr(30040))
+        debuglog("TAG-GEN: Manual arg received, exiting after single execution.")
+        sys.exit(0)
+    elif wipeout:
+        dialog = xbmcgui.Dialog()
+        ok = dialog.ok("Tag Generator", _getstr(30034) + str(wipealltags()) + _getstr(30035))
+        debuglog("TAG-GEN: Wipeout arg received, exiting after single execution.")
+        sys.exit(0)
+    else:
+        xbmc.log(msg="TAG-GEN: Sleeping for " + str(c_refresh) + " hours", level=xbmc.LOGNOTICE)
+        while not monitor.abortRequested():
+            if (datetime.datetime.now() - start_time).total_seconds() < service_interval:
+                xbmc.sleep(micronap)
```

Comment:
use xbmc.Monitor.waitForAbort() while sleeping in long loops

Reply:
tested and implemented - been meaning to do that for a while; thanks.